src/main/java/com/nutomic/syncthingandroid/activities/FolderActivity.java

```diff
@@ -460,4 +425,64 @@ private Dialog createDiscardDialog() {
                 .setNegativeButton(android.R.string.cancel, null)
                 .create();
     }
+
+    @Override
+    public void onDialogClosed(Bundle arguments) {
+        updateVersioning(arguments);
+    }
+
+    private void updateVersioning(Bundle arguments) {
+        String type = arguments.getString("type");
+        arguments.remove("type");
+
+        if (type.equals("none")){
+            mFolder.versioning = new Folder.Versioning();
+        }else {
+            mFolder.versioning.type = type;
+
+            for (String key : arguments.keySet()) {
+                mFolder.versioning.params.put(key, arguments.getString(key));
+            }
+        }
+        mVersioningDescriptionView.setText(getVersioningDescription());
+        mFolderNeedsToUpdate = true;
+    }
+
+    private String getVersioningDescription() {
+        String description = getString(R.string.none);
+
+        if (mFolder == null || mFolder.versioning.type == null){
+            return description;
+        }
+
+        switch (mFolder.versioning.type) {
+            case "simple":
+                description = getString(R.string.type) + ": " + getString(R.string.simple) + "\n"
+                        + getString(R.string.keep_versions) +": " + mFolder.versioning.params.get("keep");
+                break;
+            case "trashcan":
+                description = getString(R.string.type) + ": " + getString(R.string.trashcan) + "\n"
+                        + getString(R.string.clean_out_after) +": " + mFolder.versioning.params.get("cleanoutDays");
+                break;
+            case "staggered":
+                int maxAge = getMaxAgeInDays();
+                description = getString(R.string.type) + ": " + getString(R.string.staggered) + "\n"
+                        + getString(R.string.maximum_age) + ": " + (maxAge == 0 ? getString(R.string.never_delete) : maxAge) + "\n"
+                        + getString(R.string.versions_path)+ ": " + mFolder.versioning.params.get("versionsPath");
+                break;
+            case "external":
+                description = getString(R.string.type) + ": " + getString(R.string.external) + "\n"
+                        + getString(R.string.command) + ": " + mFolder.versioning.params.get("command");
```

Comment:
You will get the correct translation, but lint should  show a warning if you do string concatenation in code like that, probably because other languages might have the text in a different order. You can run `./gradlew lint` on command line to check, though it might not catch it because its not assigned directly to a TextView.